apps/wallet-mobile/src/yoroi-wallets/cardano/byron/ByronWallet.ts

```diff
@@ -1007,10 +1007,29 @@ export class ByronWallet implements YoroiWallet {
 
     await this.utxoManager.sync(addresses)
 
-    this._utxos = await this.utxoManager.getCachedUtxos()
+    const newUtxos = await this.utxoManager.getCachedUtxos()
 
-    // notifying always -> sync from lib need to flag if something has changed
-    this.notify({type: 'utxos', utxos: this.utxos})
+    if (this.hasUtxoUpdated(this._utxos, newUtxos)) {
+      this._utxos = newUtxos
+
+      this.notify({type: 'utxos', utxos: this.utxos})
+    }
+  }
+
+  private hasUtxoUpdated(oldUtxos: RawUtxo[], newUtxos: RawUtxo[]): boolean {
+    if (oldUtxos.length !== newUtxos.length) {
+      return true
+    }
+
+    const oldUtxoIds = new Set(oldUtxos.map((utxo) => utxo.utxo_id))
```

Comment:
How often do we call hasUtxoUpdated ?  Creating a set from the scratch for each function call can be resource consuming